table.cc

```diff
@@ -2446,6 +2449,30 @@ table::run_with_compaction_disabled(std::function<future<> ()> func) {
     });
 }
 
+bool
+table::set_auto_compaction(bool enabled) {
+    if (enabled) {
+        if (_compaction_disabled_by_user) {
+            _compaction_disabled_by_user = false;
+            if (--_compaction_disabled == 0) {
+                // we're turning if on again, use function that does not increment
+                // the counter further.
+                do_trigger_compaction();
+            }
+            return true;
+        }
+    } else {
+        if (!_compaction_disabled_by_user) {
+            ++_compaction_disabled;
+            _compaction_disabled_by_user = true;
+            (void) _compaction_manager.remove(this);
+            return true;
+        }
+    }
+
+    return false;
```

Comment:
What does this returned value mean?

Reply:
I thought to allow API to return a value if it made a distinct change. But now I am thinking its unnecessary. I will remove it.

Reply:
removed